queries.sql

```diff
@@ -64,4 +64,25 @@ SELECT neutered, SUM(escape_attempts) AS total_escapes FROM animals GROUP BY neu
 SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight FROM animals GROUP BY species;
 
 -- Average number of escape attempts per animal type of those born between 1990 and 2000
-SELECT species, AVG(escape_attempts) AS avg_num_of_escape FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31' GROUP BY species;
\ No newline at end of file
+SELECT species, AVG(escape_attempts) AS avg_num_of_escape FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31' GROUP BY species;
+
+-- What animals belong to Melody Pond?
+SELECT a.name FROM animals a JOIN owners b ON a.owner_id = b.id WHERE b.full_name = 'Melody Pond';
+
+-- List of all animals that are pokemon (their type is Pokemon).
+SELECT a.name FROM animals a JOIN species s ON a.species_id = s.id WHERE s.name = 'Pokemon';
+
+-- List all owners and their animals, remember to include those that don't own any animal.
+SELECT o.full_name, COALESCE(array_agg(a.name), '{}'::VARCHAR[]) AS owned_animals FROM owners o LEFT JOIN animals a ON o.id = a.owner_id GROUP BY o.full_name;
+
+-- How many animals are there per species?
+SELECT s.name, COUNT(*) AS animal_count FROM species s JOIN animals a ON s.id = a.species_id GROUP BY s.name;
+
+-- List all Digimon owned by Jennifer Orwell.
+SELECT a.name FROM animals a JOIN species s ON a.species_id = s.id JOIN owners o ON a.owner_id = o.id WHERE s.name = 'Digimon' AND o.full_name = 'Jennifer Orwell';
+
```

Comment:
I'd suggest that the two JOINS here should also be a LEFT JOIN ,something like this :

   SELECT animals.* FROM animals
   LEFT JOIN owners
   ON animals.owner_id = owners.id
   LEFT JOIN species
   ON animals.species_id = species.id
   WHERE owners.full_name = 'Jennifer Orwell'
   AND species.name = 'Digimon';